### PR TITLE
Updates for nng-based faabric branch

### PR DESCRIPTION
**Pull request by eigenraven**
In short: thread-safe sockets to reduce socket creation inside worker threads. See <https://github.com/faasm/faabric/pull/286> for all the details.
Requires the faabric PR to be merged first to update the submodule pointer.

### CODE REVIEW

**Review by csegarragonz — APPROVED** [review]
LGTM after faabric's PR is merged in.

This is quite a big change, with potentially performance differences. I wonder if it is worth making a new minor Faasm tag :thinking: 

@eigenraven do you have all the permissions to trigger a tag upgrade?

**Review by Shillaker — APPROVED** [review]

**Comment by csegarragonz** [comment]
@eigenraven FYI I have merged faabric, updated the submodule, and bumped the code version to trigger a new tag build.

Will merge the PR in after all checks pass.

**Pull request merged**
